source/tests/system/nirfmxspecan_driver_api_tests.cpp

```diff
@@ -424,6 +445,200 @@ TEST_F(NiRFmxSpecAnDriverApiTests, BuildSpurString_ReturnsSpurString)
+
+template <typename T>
+inline std::vector<T> vector_concat(const std::vector<T>& first, const std::vector<T> second)
+{
+  auto result = std::vector<T>(first);
+  result.insert(result.end(), second.cbegin(), second.cend());
+  return result;
+}
+
+// These will initialize as separate driver resource handles.
+const auto INDEPENDENT_RESOURCES = std::vector<ResourcePair>{
+    {"specan1", "specan2"},
+    {"specan1,specan2", "specan3,specan4"},
+    {"", ""},
+    {"specan1", ""},
+};
+
+// These will initialize as the same driver resource handle, the second
+// session will increment that resource's ref count.
+const auto DUPLICATE_RESOURCES = std::vector<ResourcePair>{
+    {"specan1", "specan1"},
+    {"specan1,specan2", "specan2,specan1"},
+    {"specan1,specan2", "specan1,specan2"},
+    {"specan1,specan2,specan3", "specan3,specan1,specan2"},
+};
+
+// All of these are valid to initialize as separate grpc-device sessions at the same time.
+// Internally, they may or may not be separate driver handles, but they will each have their
+// own ref count.
+const auto COMPATIBLE_RESOURCES = vector_concat(INDEPENDENT_RESOURCES, DUPLICATE_RESOURCES);
+
+// These are conflicting because the second set overlaps the first set but IS NOT IDENTICAL.
+const auto CONFLICTING_RESOURCES = std::vector<ResourcePair>{
+    {"specan1", "specan1,specan2"},
+    {"specan1,specan2", "specan1"},
+    {"specan1,specan2", "specan1,specan2,specan3"},
+};
+
+using NiRFmxSpecAnDriverApiCompatibleResourceInitTests = NiRFmxSpecAnDriverApiResourceInitTests;
+INSTANTIATE_TEST_SUITE_P(
+    CompatibleResourceInitTestCases,
+    NiRFmxSpecAnDriverApiCompatibleResourceInitTests,
+    ValuesIn(COMPATIBLE_RESOURCES),
+    ResourceTuplePrinter{});
+
+TEST_P(NiRFmxSpecAnDriverApiCompatibleResourceInitTests, InitializeTwoCompatibleResourcesAsSeparateSessions_CloseEachSession_EachIsClosedIndependently)
+{
+  const auto session1 = init_session(stub(), PXI_5663, std::get<0>(GetParam()));
+  const auto session2 = init_session(stub(), PXI_5663, std::get<1>(GetParam()));
+  EXPECT_NE(session1.id(), session2.id());
+  EXPECT_VALID_DRIVER_SESSION(session1);
+  EXPECT_VALID_DRIVER_SESSION(session2);
+
+  close_session(stub(), session1);
+  EXPECT_INVALID_DRIVER_SESSION(session1);
+  EXPECT_VALID_DRIVER_SESSION(session2);
+  close_session(stub(), session2);
+  EXPECT_INVALID_DRIVER_SESSION(session2);
+}
+
+// Same as NiRFmxSpecAnDriverApiSuccessfulResourceInitTests, but with 3 sessions instead of the standard 2.
+TEST_F(NiRFmxSpecAnDriverApiTests, InitializeThreeIdenticalResourcesAsSeparateSessions_CloseEachSession_EachIsClosedIndependently)
+{
+  const auto session1 = init_session(stub(), PXI_5663, "specan1,specan2");
+  const auto session2 = init_session(stub(), PXI_5663, "specan2,specan1");
+  const auto session3 = init_session(stub(), PXI_5663, "specan1,specan2");
+  EXPECT_NE(session1.id(), session2.id());
+  EXPECT_NE(session1.id(), session3.id());
```

Comment:
Should we go ahead and add `EXPECT_NE(session2.id(), session3.id())`? Just to make sure we're not getting {1, 2, 2} or something for the IDs.